op-supervisor/supervisor/backend/db/db.go

```diff
@@ -49,6 +59,75 @@ func (db *ChainsDB) Resume() error {
 	return nil
 }
 
+// UpdateCrossSafeHeads updates the cross-heads of all chains
+// this is an example of how to use the SafetyChecker to update the cross-heads
+func (db *ChainsDB) UpdateCrossSafeHeads() error {
+	checker := NewSafetyChecker(Safe, *db)
+	return db.UpdateCrossHeads(checker)
+}
+
+// UpdateCrossHeadsForChain updates the cross-head for a single chain.
+// the provided checker controls which heads are considered.
+// TODO: we should invert control and have the underlying logDB call their own update
+// for now, monolithic control is fine. There may be a stronger reason to refactor if the API needs it.
+func (db *ChainsDB) UpdateCrossHeadsForChain(chainID types.ChainID, checker SafetyChecker) error {
```

Comment:
It would be nice to encapsulate this for a specific chain, so we can run it easily in parallel. This is fine for now though, we can refactor later if needed.

Reply:
This function does update only a single chain. There's a function just below it called `UpdateCrossHeads` which loops over all known chains and calls this update function.